src/sentry/workflow_engine/endpoints/serializers/workflow_group_history_serializer.py

```diff
@@ -128,11 +128,14 @@ def fetch_workflow_groups_paginated(
         .annotate(detector_id=Subquery(group_max_dates.values("detector_id")))
     )
 
+    # Count distinct groups for pagination
+    group_count = qs.count()
```

Comment:
just a quick note that this will add another query and may impact performance; i'm guessing we need this to get the total number of results compared to the paginated list though.